zkevm-circuits/src/evm_circuit/witness.rs

```diff
@@ -24,19 +26,17 @@ pub struct Block<F> {
 
 #[derive(Debug, Default)]
 pub struct Transaction<F> {
-    // Context
     pub id: usize,
     pub nonce: u64,
     pub gas: u64,
-    pub gas_tip_cap: Word,
-    pub gas_fee_cap: Word,
+    pub gas_price: Word,
```

Comment:
so where to handle gas_limit of tx ?

Reply:
Do you mean checking if block's gas pool is still enough for this tx? If so, I forgot that in the spec and I will add that asap, then we can implement it in the further PRs.

Reply:
yea, that is . on the other hand, each transaction have gas limit setting when making by user, if a transaction successfully, need to constrain consumed  gas value <= gas limit

Reply:
The checking of tx gas limit is not at the beginning (we charge first no matter it succeeds or not to avoid any kinds of DoS).

Ignore EIP-1559, a valid tx in a block must pass:

1. The block gas pool has sufficient `gas` for the `gas` specified in tx.
2. The signer recovered from signature has enough balance `gas * gas_price` to buy the `gas`.
3. The `gas` is enough for intrinsic usage, that is `53000 if is_create else 21000 + call_data_gas_cost`
4. ... (not gas related)

Then the next step will have `gas_left` set to `gas - intrinsic_gas_cost`. Does it make sense in this way?

Reply:
sounds good , the tx table itself should have corresponding  circuit constraint like byte code table, right ?

Reply:
Yes, if we decide to instantiate the `tx_table` by something like Tx circuit, then we can verify `call_data_gas_cost` is correct by some constraints (should be super easy to implement).

But currently we just let verifier build everything directly (including calculating `call_data_gas_cost` and `hash`) and instantiate the `tx_table` with only instance columns, without additional advice columns or constraint enforced.